svid/x509svid/src/svid.c

```diff
@@ -352,8 +346,8 @@ void x509svid_SVID_Free(x509svid_SVID *svid)
 
 x509svid_SVID *x509svid_SVID_GetDefaultX509SVID(x509svid_SVID **svids)
 {
-    if(!svids) {
-        return NULL;
+    if(arrlenu(svids) > 0) {
```

Comment:
Like this one but not sure. Looks cleaner and means the same, but if we get a stale pointer it MAY crash here and not on array access outside. It might actually be better, if we want to crash on "get" instead of later.
In that case, should we verify the SVID for integrity/validity, before returning? or is it supposed to be validated later? (this goes for the old and new implementations).
In any case, it does look cleaner.

Reply:
Weel, the function should expect a stb array of x509 svid object pointers. If the variable 'svids' is NULL, 'arrlenu' already deals with that. If 'svids' is not NULL and it is not a stb array either, 'arrlenu' will actually read garbage, but is unlikely to crash. Btw, will this function be used for non stb arrays?